### PR TITLE
Make code to be compatible with Symfony 2.4+

### PR DESCRIPTION
**Pull request by stloyd**
Replaces #1017, #1019, #1021

### CODE REVIEW

**Comment by XWB** [comment]
Looks good.

File: Subscriber/PaginateElasticaQuerySubscriber.php [hunk 1]
```diff
@@ -2,19 +2,30 @@
...
+    /**
+     * @param RequestStack|Request $requestStack
+     */
+    public function setRequest($requestStack)
     {
-        $this->request = $request;
+        if ($requestStack instanceof Request) {
+            $this->request = $requestStack;
+        } elseif ($requestStack instanceof RequestStack) {
+            $this->request = $requestStack->getMasterRequest();
```
Comment (rybakit) [inline Subscriber/PaginateElasticaQuerySubscriber.php]: It will not work for sub-requests, right? Don't know if it's important though.

Comment (jvandesande) [inline Subscriber/PaginateElasticaQuerySubscriber.php]: It has never worked for sub requests. So it would be backwards incompatible if it would work now.

Comment (XWB) [inline Subscriber/PaginateElasticaQuerySubscriber.php]: Other bundles use `getCurrentRequest()`.

Comment (stloyd) [inline Subscriber/PaginateElasticaQuerySubscriber.php]: `getCurrentRequest()` can return `null`.

Comment (rybakit) [inline Subscriber/PaginateElasticaQuerySubscriber.php]: @stloyd  The `Request` instance can be null as well: http://symfony.com/doc/2.3/cookbook/service_container/scopes.html#using-a-synchronized-service

**Pull request merged**